pkg/api/kubegraph/analysis/hpa.go

```diff
@@ -0,0 +1,178 @@
+package analysis
+
+import (
+	"fmt"
+	"strings"
+
+	"k8s.io/kubernetes/pkg/apis/extensions"
+
+	graphapi "github.com/gonum/graph"
+	"github.com/gonum/graph/path"
+
+	osgraph "github.com/openshift/origin/pkg/api/graph"
+	"github.com/openshift/origin/pkg/api/kubegraph"
+	kubenodes "github.com/openshift/origin/pkg/api/kubegraph/nodes"
+	deploygraph "github.com/openshift/origin/pkg/deploy/graph"
+	deploynodes "github.com/openshift/origin/pkg/deploy/graph/nodes"
+)
+
+const (
+	// HPAMissingScaleRefError denotes an error where a Horizontal Pod Autoscaler does not have a reference to an object to scale
+	HPAMissingScaleRefError = "HPAMissingScaleRef"
+	// HPAMissingCPUTargetError denotes an error where a Horizontal Pod Autoscaler does not have a CPU target to scale by.
+	// Currently, the only supported scale metric is CPU utilization, so without this metric an HPA is useless.
+	HPAMissingCPUTargetError = "HPAMissingCPUTarget"
+	// HPAOverlappingScaleRefWarning denotes a warning where a Horizontal Pod Autoscaler scales an object that is scaled by some other object as well
+	HPAOverlappingScaleRefWarning = "HPAOverlappingScaleRef"
+)
+
+// FindHPASpecsMissingCPUTargets scans the graph in search of HorizontalPodAutoscalers that are missing a CPU utilization target.
+// As of right now, the only metric that HPAs can use to scale pods is the CPU utilization, so if a HPA is missing this target it
+// is effectively useless.
+func FindHPASpecsMissingCPUTargets(graph osgraph.Graph, namer osgraph.Namer, patchCommandName string) []osgraph.Marker {
```

Comment:
Well, actually in most commands, we are not passing down names. Those two funcs are the exception. I see why we would want to pass the actual binary name (oc vs openshift cli) but that depends on how much we care about it. People will most probably just run oc.